src/Makefile.am

```diff
@@ -231,6 +231,7 @@ BITCOIN_CORE_H = \
   util/vector.h \
   validation.h \
   validationinterface.h \
+  variant.h \
```

Comment:
This seems consistent since our `optional` and `span` replacements are also at the top level. But I wonder if they should all move to `src/util/` or some place to keep the top level less crowded. Probably something to deal with in a different PR

Reply:
See also #15732